### PR TITLE
[Merged by Bors] - Fixed docs for `derive(WorldQuery)`.

### PR DESCRIPTION
**Pull request by CGMossa**
For `derive(WorldQuery)`, there are three structs generated, `Item`, `Fetch` and `State`. 
These inherit the visibility of the derived structure, thus `#![warn(missing_docs)]` would
warn about missing documentation for these structures.

- [ ] I'd like some advice on what to write here, as I personally don't really understand `Fetch` nor `State`.

### CODE REVIEW

File: crates/bevy_ecs/macros/src/fetch.rs
Comment (alice-i-cecile) [inline crates/bevy_ecs/macros/src/fetch.rs:169]: "Automatically generated [`WorldQuery`] item type for [`Foo`], returned when iterating over query results."

File: crates/bevy_ecs/macros/src/fetch.rs
Comment (alice-i-cecile) [inline crates/bevy_ecs/macros/src/fetch.rs:264]: "Automatically generated internal [`WorldQuery`] state type for [`Foo`], used for caching."

**Comment by Nilirad** [comment]
I thought that `#[automatically_derived]` items wouldn't trigger a warning.

File: crates/bevy_ecs/macros/src/fetch.rs
Comment (CGMossa) [reply crates/bevy_ecs/macros/src/fetch.rs:178]: For all things or maybe only for `Fetch`?

Comment (alice-i-cecile) [reply crates/bevy_ecs/macros/src/fetch.rs]: IMO we should hide both the fetch and state types; they're not going to be useful to end users.

**Comment by CGMossa** [comment]
Done. Sorry :( I cannot believe I didn't notice.

**Comment by alice-i-cecile** [comment]
@BoxyUwU @james7132 are my docs for the generated structs basically fine?

**Comment by BoxyUwU** [comment]
havign docs on a doc(hidden) thing seems weird to me. the docs on the item type I think would be wrong for the `ReadOnly` version as it would link to the wrong worldquery?

**Comment by CGMossa** [comment]
I added a suggestion; I don't know if it improves things or not :(

**Comment by CGMossa** [comment]
@BoxyUwU have you seen what I've suggested we change it to?
Hopefully this works for you.

**Comment by BoxyUwU** [comment]
I'm not sure what you mean by "suggested"- anyway the docs look fine to me but it seems like no commits were added? Whoops I must have misread the docs last time sorry

**Comment by CGMossa** [comment]
Do you not see this

**Comment by alice-i-cecile** [comment]
Nope; you need to finish your review. That's what "pending" means.

**Comment by CGMossa** [comment]
I've messed this up. I was waiting for Boxy, but it turns out that I was the one creating confusion. Here's what I suggested. The thing is that the link to `[`ReadOnlyWorldQuery`]` doesn't seem to work.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by bzm3r** [comment]
@CGMossa I think Alice is fine with it as is. Could you resolve conflicts?

**Review by bzm3r — APPROVED** [review]

**Comment by CGMossa** [comment]
Ah.. I have to rebase.. I'll see if I get to it tomorrow.

**Comment by CGMossa** [comment]
Please @bzm3r to look at the changes once more, and see if something was lost the messy rebase I did on this.

**Comment by CGMossa** [comment]
Don't merge this please; Testing if it serves it purpose shows that something is missing:

```
(base) PS C:\Users\angus\Documents\GitHub\bevy_fresh> cargo check --examples
    Blocking waiting for file lock on build directory
    Checking bevy v0.9.0-dev (C:\Users\angus\Documents\GitHub\bevy_fresh)
error: missing documentation for a struct
  --> examples/ecs/custom_query_param.rs:41:1
   |
41 | pub struct ReadOnlyCustomQuery<T: Component + Debug, P: Component + Debug> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> examples/ecs/custom_query_param.rs:14:11
   |
14 | #![forbid(missing_docs)]
   |           ^^^^^^^^^^^^

error: could not compile `bevy` due to previous error
```

**Comment by CGMossa** [comment]
This is now fixed.

**Pull request closed without merging**
